### PR TITLE
feat(tile-select): Add disabled styling for tile select, updates hover and focus style

### LINKED ISSUE #1063: Enhancement: Clearer disabled state for calcite-tile-select
### Description
The disabled state for calcite-title-select is not very clear. It doesn't look any different except on hover when there is a slight color difference. I've created a codepen to show the issue: https://codepen.io/noahmulfinger/pen/LYNwGZw?editors=1000

### Acceptance Criteria 
There is a clear clear disabled state for calcite-tile-select.

#### Which Component
calcite-tile-select

### LINKED ISSUE #1098: Research: better selection UX for large visual checkbox tile selects
so everything seems to be good except for these. maybe a future enhancement since it's a weird issue with the hover/focus

Really hard to tell when deselecting something

image

_Originally posted by @bstifle in https://github.com/Esri/calcite-components/pull/1072#issuecomment-705110452_

### PR DESCRIPTION
**Pull request by macandcheese**
**Related Issue:** 

Resolves #1063 cc @noahmulfinger 
Resolves #1098 cc @eriklharper @bstifle 

Adds disabled styling:

Screen Shot 2020-11-04 at 10 58 48 AM

While adding the disabled styling, I ran into the issues described in #1098, so I tried to alleviate that as much as possible by simplifying the styling. I tried the "outset" focus as @bstifle suggested but it was ultimately unreliable with similar weirdness.

For tile-select items that SHOW a radio or checkbox, the focus state is applied to that input within the tile. Hover state for these configurations is a darker border.

For tile-select items that DO NOT SHOW a radio or checkbox, we apply an inset focus style. Hover state for these configurations is a blue border.

Additionally I've lightened the text for header and description by one step when a wrapped calcite-tile is NOT active, so that on hover, focus, or active, the text for header and description can darken by one step as a better indicator. This also helps the "focusing over a checkbox without a radio or checkbox but it's not selected" visual. That's still a weak point IMO but, not sure what else we can do there.

| Current | This PR |
| --- | --- |
|Screencast 2020-11-04 at 9 43 41 AM (1) | Screencast 2020-11-04 at 10 44 03 AM |

### CODE REVIEW

File: src/components/calcite-tile-select/calcite-tile-select.scss [hunk 1]
```diff
@@ -14,49 +14,40 @@ $spacing: $baseline/2;
   position: relative;
   transition: $transition;
   vertical-align: top;
+  @apply z-10;
+}
+
+:host(:hover) {
+  @apply z-20;
+  box-shadow: 0 0 0 1px var(--calcite-ui-border-1);
```
Comment (paulcpederson) [inline src/components/calcite-tile-select/calcite-tile-select.scss:22]: does this work in RTL?

**Review by paulcpederson — APPROVED** [review]

**Comment by paulcpederson** [comment]
Had a question, but LGTM otherwise.

File: src/components/calcite-tile-select/calcite-tile-select.scss (hunk 1 above)
Comment (macandcheese) [reply src/components/calcite-tile-select/calcite-tile-select.scss:22]: I think box shadow spread would be the same, but looking at the component there aren't any RTL considerations for rendering content or styling. I'll open a separate issue for that. The nomenclature for "show-input" should also be changed to "start" and "end"

Screen Shot 2020-11-05 at 11 08 36 AM

Comment (macandcheese) [reply src/components/calcite-tile-select/calcite-tile-select.scss:22]: Follow up issue for RTL support: https://github.com/Esri/calcite-components/issues/1230

**Comment by bstifle** [comment]
image
cant get local to run, but this look good. love the inset @macandcheese

**Review by bstifle — APPROVED** [review]

**Review by eriklharper — APPROVED** [review]
Rad!

**Comment by bstifle** [comment]
ok this is the only thing im seeing. @macandcheese 

UX is bad, cant tell when they are unchecked. can we just make the border go back to gray

image
image

**Comment by macandcheese** [comment]
Merging, @bstifle can you open an enhancement w proposed design for that since it's basically the same as what's in master now.

**Pull request merged**